### PR TITLE
Delete book endpoint added

### PR DESCRIPTION
**Pull request by bokimilinkovic**
https://trello.com/c/e54L4fBV/64-add-delete-book-endpoint

### CODE REVIEW

File: repository/mock/database.go [hunk 1]
```diff
@@ -89,3 +89,14 @@ func (db *DB) Update(toUpdate *model.Book) error {
...
+func (db *DB)Delete(toDelete *model.Book)error{
+	temp := db.books
+	db.books=[]model.Book{}
+	for _, book := range temp{
+		if book.Id != toDelete.Id{
+			db.books=append(db.books,book)
+		}
+	}
+	return nil
+}
```
Comment (sfqi) [inline repository/mock/database.go]: Maybe you could use `findBookByID` to get the location and then do something like [this](https://stackoverflow.com/questions/37334119/how-to-delete-an-element-from-a-slice-in-golang#answer-55381756) or [this](https://stackoverflow.com/questions/37334119/how-to-delete-an-element-from-a-slice-in-golang#answer-37335777).

**Comment by sfqi** [comment]
Tests?

File: repository/mock/database.go (hunk 1 above)
Comment (sfqi) [reply repository/mock/database.go]: Do we need `temp` and `db.books=[]model.Book{}`?

File: handler/handlers.go [hunk 2]
```diff
@@ -204,6 +204,27 @@ func (b *BookHandler) Get(w http.ResponseWriter, r *http.Request) {
...
+	}
+	book, err := b.Db.FindBookByID(id)
+	if err != nil {
+		errorFindingBook(w, err)
+		return
+	}
+	if err := b.Db.Delete(book); err != nil {
+		return
+	}
+	fmt.Fprintln(w,"Book successfuly deleted")
```
Comment (sfqi) [inline handler/handlers.go]: Just return `StatusNoContent`, no need to print anything

File: repository/mock/database.go (hunk 1 above)
Comment (sfqi) [reply repository/mock/database.go]: Also, let's try to be consistent with empty spaces (as in, we should use them)

File: handler/handlers.go (hunk 2 above)
Comment (bokimilinkovic) [reply handler/handlers.go]: you mean just to write fmt.Fprintln(w, http.SatusNoContent)?  I cant return anything in handler

Comment (sfqi) [reply handler/handlers.go]: You figured it out in the meantime.

File: repository/mock/database.go (hunk 1 above)
Comment (sfqi) [reply repository/mock/database.go]: What about these comments?

File: repository/mock/database.go [hunk 3]
```diff
@@ -89,3 +89,16 @@ func (db *DB) Update(toUpdate *model.Book) error {
 	db.books[index] = *book
 	return nil
 }
+
+func (db *DB)Delete(book *model.Book)error{
+	_,loc,err:=db.findBookByID(book.Id)
+	if err != nil{
+		return err
+	}
+	db.books=remove(db.books,loc)
```
Comment (sfqi) [inline repository/mock/database.go]: `db.books = append(db.books[:loc], db.books[loc+1:]...)` would have worked as well, I think.

File: handler/handlers.go [hunk 4]
```diff
@@ -204,6 +204,27 @@ func (b *BookHandler) Get(w http.ResponseWriter, r *http.Request) {
...
+		errorConvertingId(w, err)
+		return
+	}
+	book, err := b.Db.FindBookByID(id)
+	if err != nil {
+		errorFindingBook(w, err)
+		return
+	}
+	if err := b.Db.Delete(book); err != nil {
+		return
```
Comment (sfqi) [inline handler/handlers.go]: No error handling here?

**Pull request merged**